ql/src/test/queries/clientpositive/udf_trim.q

```diff
@@ -1,2 +1,20 @@
 DESCRIBE FUNCTION trim;
 DESCRIBE FUNCTION EXTENDED trim;
+
+SELECT '"' || trim('   tech   ') || '"';
+
+SELECT '"' || TRIM(' '  FROM  '   tech   ') || '"';
+
+SELECT '"' || TRIM(LEADING '0' FROM '000123') || '"';
+
+SELECT '"' || TRIM(TRAILING '1' FROM 'Tech1') || '"';
+
+SELECT '"' || TRIM(BOTH '1' FROM '123Tech111') || '"';
+
+SELECT '"' || ltrim('   tech   ') || '"', '"' || rtrim('   tech   ') || '"';
+
+SELECT '"' || lTRIM('0'  FROM  '000123') || '"', '"' || rTRIM('0'  FROM  '000123') || '"';
+
+SELECT trim('000123', '0');
```

Comment:
could you also add some`null` cases as well `trim(null,'x')` and `trim('x',null)`

I know they will probably work okay; but its better to cover them

Reply:
`null` cannot be passed as a parameter to `trim`. It works in Postgres and Oracle but not supported in Hive.
The following was executed on master without this patch:
```
select trim(null);

 org.apache.hadoop.hive.ql.parse.SemanticException: Line 6:7 Wrong arguments 'TOK_NULL': trim takes only STRING/CHAR/VARCHAR types. Found VOID
```

IMHO we can implement this in a follow-up patch.